### PR TITLE
Adding search history app

### PR DESCRIPTION
**Pull request by OfirBachar**
Adding search history app
Create SearchHistory model

At the time of creating this PR we still don't have the Client and Professional apps.

The model has  foreign keys from client and professional which have not been merged yet, so for now it was added within a comment.

When we add the Client and Professional apps, I'll make the necessary changes in the models.py.

Resolve https://github.com/redhat-beyond/Pro4U/issues/37

### CODE REVIEW

**Review by idoyek — APPROVED** [review]
LGTM

**Review by guybeckenstein — APPROVED** [review]
LGTM

File: SearchHistory/models.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+from django.utils import timezone
+# from client.models import Client
+# from professional.models import Professional
+
+
+class SearchHistory(models.Model):
+    SearchID = models.BigAutoField(primary_key=True)
+    # ProfessionalID = models.ForeignKey(Professional, on_delete=models.CASCADE)
+    # ClientID = models.ForeignKey(Client, on_delete=models.CASCADE)
+    Date = models.DateTimeField(default=timezone.now)
```
Comment (idosinger98) [inline SearchHistory/models.py:11]: The variable names are not correct, please follow the following document https://peps.python.org/pep-0008/

Comment (OfirBachar) [reply SearchHistory/models.py]: fixed

**Review by Yarboa — CHANGES_REQUESTED** [review]
Please update with test data files to populate relevant data
Till Pro/Client models are merged to main, you can use string variables with search model

**Comment by OfirBachar** [comment]
> Please update with test data files to populate relevant data Till Pro/Client models are merged to main, you can use string variables with search model

done

**Comment by OfirBachar** [comment]
> Please update with test data files to populate relevant data Till Pro/Client models are merged to main, you can use string variables with search model

done

**Review by patrisiakaplun — APPROVED** [review]
LGTM

File: SearchHistory/models.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from django.db import models
+from django.utils import timezone
+# from client.models import Client
+# from professional.models import Professional
+
+
+class SearchHistory(models.Model):
+    search_ID = models.BigAutoField(primary_key=True)
+    # professional_ID = models.ForeignKey(Professional, on_delete=models.CASCADE)
```
Comment (Yarboa) [inline SearchHistory/models.py:9]: Please remove this and open an Issue
Once update models once #44 is merged

File: SearchHistory/tests.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# from django.test import TestCase
```
Comment (Yarboa) [inline SearchHistory/tests.py:1]: Please remove this file, it is empty

Comment (Yarboa) [reply SearchHistory/tests.py:1]: Also, please rebase and merge conflicts

**Review by idosinger98 — COMMENTED** [review]
Please create a function that receives as input Client_id and returns as output the last five professionals he searched for

File: SearchHistory/migrations/0002_test_data.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        from SearchHistory.models import SearchHistory
+
+        test_data = [
+            ('1', 'Test Pro1', 'Test Client1'),
+            ('2', 'Test Pro2', 'Test Client2'),
+            ('3', 'Test Pro3', 'Test Client3'),
+            ('4', 'Test Pro4', 'Test Client4'),
+        ]
+
+        with transaction.atomic():
```
Comment (dkeler) [inline SearchHistory/migrations/0002_test_data.py:19]: What is the reason for using an `atomic` transaction here?

Comment (Yarboa) [reply SearchHistory/migrations/0002_test_data.py:19]: @dkeler 
https://docs.djangoproject.com/en/4.2/howto/writing-migrations/#non-atomic-migrations

**Comment by patrisiakaplun** [comment]
@OfirBachar squash?

**Comment by Yarboa** [comment]
@OfirBachar  Please rebase

**Comment by OfirBachar** [comment]
> @OfirBachar squash?

Done @patrisiakaplun

**Comment by OfirBachar** [comment]
> @OfirBachar Please rebase

Done @Yarboa

**Review by patrisiakaplun — APPROVED** [review]
LGTM

**Review by idoyek — APPROVED** [review]
LGTM

**Review by idosinger98 — APPROVED** [review]
LGTM

**Review by Yarboa — APPROVED** [review]
LGTM

**Pull request merged**